ui/revenuecatui/src/test/java/com/revenuecat/purchases/ui/revenuecatui/data/processed/TemplateConfigurationFactoryTest.kt

```diff
@@ -0,0 +1,89 @@
+
+    private val template2Configuration = TemplateConfigurationFactory.create(
+        paywallViewMode,
+        TestData.template2,
+        listOf(TestData.Packages.weekly, TestData.Packages.monthly, TestData.Packages.annual),
+        emptySet(),
+        Locale.US,
+    )
+
+    @Test
+    fun `template configuration has correct template`() {
+        assertThat(template2Configuration.template).isEqualTo(PaywallTemplate.TEMPLATE_2)
+    }
+
+    @Test
+    fun `template configuration has correct view mode`() {
+        assertThat(template2Configuration.mode).isEqualTo(paywallViewMode)
+    }
+
+    @Test
+    fun `template configuration has correct paywall data configuration`() {
+        assertThat(template2Configuration.configuration).isEqualTo(TestData.template2.config)
+    }
+
+    @Test
+    fun `template configuration has correct package configuration`() {
+        assertThat(template2Configuration.packages).isInstanceOf(
+            TemplateConfiguration.PackageConfiguration.Multiple::class.java
+        )
+        val packageConfiguration = template2Configuration.packages as TemplateConfiguration.PackageConfiguration.Multiple
+
+        val expectedConfiguration = TemplateConfiguration.PackageConfiguration.Multiple(
+            first = getPackageInfo(TestData.Packages.weekly),
+            default = getPackageInfo(TestData.Packages.monthly),
+            all = listOf(
+                getPackageInfo(TestData.Packages.weekly),
+                getPackageInfo(TestData.Packages.monthly),
+                getPackageInfo(TestData.Packages.annual)
+            ),
+        )
+        assertThat(packageConfiguration).isEqualTo(expectedConfiguration)
+    }
+
+    @Test
+    fun `template configuration has correct images`() {
+        assertThat(template2Configuration.images.iconUri)
+            .isEqualTo(Uri.parse("https://assets.pawwalls.com/9a17e0a7_1689854430..jpeg"))
+        assertThat(template2Configuration.images.backgroundUri)
+            .isEqualTo(Uri.parse("https://assets.pawwalls.com/9a17e0a7_1689854342..jpg"))
+        assertThat(template2Configuration.images.headerUri)
+            .isEqualTo(Uri.parse("https://assets.pawwalls.com/9a17e0a7_1689854430..jpeg"))
+    }
+
+    private fun getPackageInfo(rcPackage: Package): TemplateConfiguration.PackageInfo {
+        val processedLocalization = ProcessedLocalizedConfiguration(
+            title = "Call to action for better conversion.",
+            subtitle = "Lorem ipsum is simply dummy text of the printing and typesetting industry.",
+            callToAction = "Subscribe for {{ price_per_period }}",
+            callToActionWithIntroOffer = null,
+            offerDetails = "{{ total_price_and_per_month }}",
+            offerDetailsWithIntroOffer = "{{ total_price_and_per_month }} after {{ sub_offer_duration }} trial",
+            offerName = "{{ sub_period }}",
+            features = emptyList(),
+        )
```

Comment:
These are duplicated from `TestData`. If we change those this is going to fail. Can you reference that instead?

Reply:
I mean, once we actually process the variables, this will change. But yeah, I can remove the duplication for now.